zeroex/orderwatch/order_watcher.go

```diff
@@ -1158,89 +1158,53 @@ func (w *Watcher) convertValidationResultsIntoOrderEvents(
 		oldAmountIsMoreThenNewAmount := oldFillableAmount.Cmp(newFillableAmount) == 1
 
 		if oldFillableAmount.Cmp(big.NewInt(0)) == 0 {
-			// A previous event caused this order to be removed from DB because it's
+			// A previous event caused this order to be removed from DB because its
 			// fillableAmount became 0, but it has now been revived (e.g., block re-org
 			// causes order fill txn to get reverted). We need to re-add order and emit an event.
-			w.rewatchOrder(order, acceptedOrderInfo.FillableTakerAssetAmount, validationBlock)
+			w.rewatchOrder(order, newFillableAmount, validationBlock)
 			orderEvent := &zeroex.OrderEvent{
 				Timestamp:                validationBlock.Timestamp,
 				OrderHash:                acceptedOrderInfo.OrderHash,
 				SignedOrder:              order.SignedOrder(),
-				FillableTakerAssetAmount: acceptedOrderInfo.FillableTakerAssetAmount,
+				FillableTakerAssetAmount: newFillableAmount,
 				EndState:                 zeroex.ESOrderAdded,
 				ContractEvents:           orderHashToEvents[order.Hash],
 			}
 			orderEvents = append(orderEvents, orderEvent)
 		} else {
-			// The order is expired if ExpirationTimeSeconds is equal to or less than the timestamp
+			// The time stamp is valid if ExpirationTimeSeconds is equal to or greater than the time stamp
 			// of the validation block.
 			validationBlockTimeStampSeconds := big.NewInt(validationBlock.Timestamp.Unix())
-			isOrderExpired := order.ExpirationTimeSeconds.Cmp(validationBlockTimeStampSeconds) != 1
+			timeStampIsValid := order.ExpirationTimeSeconds.Cmp(validationBlockTimeStampSeconds) == 1
```

Comment:
This is actually incorrect. The timestamp is only valid if it less than the expiration time: https://github.com/0xProject/0x-monorepo/blob/development/contracts/exchange/contracts/src/MixinExchangeCore.sol#L171. We've had to fix this elsewhere, so I guess this slipped through the cracks. I realize that this was in here before, but let's fix it in this PR since we're changing the line.

Reply:
I think this is correct according to the code you linked? `timeStampIsValid` is true if `order.ExpirationTimeSeconds > validationBlockTimestampSeconds`?

Reply:
Whoops, I looked at this the wrong way. You’re totally right